patch/newterm.c

```diff
@@ -1,7 +1,15 @@
+extern char* argv0;
+
+static char*
+getcwd_by_pid(pid_t pid) {
+	char buf[32];
+	snprintf(buf, sizeof buf, "/proc/%d/cwd", pid);
+	return realpath(buf, NULL);
```

Comment:
Is there a reason for `realpath()` here since `chdir()` resolves symlinks and `/proc/%d/cwd` is a symlink?

Couldn't you just do this:

```
	static char buf[32];
```

And

```
	return buf;

```
You could also rename `buf` to `cwd`.

Reply:
`/proc/{pid}/cwd` is a symlink to the current working dir. If we don't use `realpath()`, then the new terminal has `$PWD` set to `/proc/{pid}/cwd` instead of the folder it is pointing to.

> I know this because I had the same thought as you. :)

Reply:
Are you sure because it works fine for me? And what could be the reason it doesn't work for you?

Reply:
You're right. It works. I might have done something else when I tested it earlier today.